From 1266d27f896f0f701a085224a29b0bb5b9a8b59e Mon Sep 17 00:00:00 2001
From: Clayton Craft <clayton@craftyguy.net>
Date: Wed, 26 Feb 2025 22:37:14 -0800
Subject: [PATCH] backend/mod: drop malloc_trim call

Apparently this is a workaround for something related to glibc?[1]
It's not supported on musl.

1. https://github.com/pop-os/cosmic-store/issues/62#issuecomment-2099358857
---
 src/backend/mod.rs | 11 -----------
 1 file changed, 11 deletions(-)

diff --git a/src/backend/mod.rs b/src/backend/mod.rs
index d1390cd..cef63ec 100644
--- a/src/backend/mod.rs
+++ b/src/backend/mod.rs
@@ -107,16 +107,5 @@ pub fn backends(locale: &str, refresh: bool) -> Backends {
         }
     });
 
-    //TODO: Workaround for xml-rs memory leak when loading appstream data
-    #[cfg(target_os = "linux")]
-    {
-        let start = Instant::now();
-        unsafe {
-            libc::malloc_trim(0);
-        }
-        let duration = start.elapsed();
-        log::info!("trimmed allocations in {:?}", duration);
-    }
-
     backends
 }
-- 
2.48.1

